### PR TITLE
Stage CNI (and harvester) images if avaliable for airgap

### PR DESCRIPTION
**Pull request by dereknola**
#### Proposed Changes ####
- For the supported CNIs, if images are found in the artifact path, automatically move to the agent images directory, removing a manual step the user would need to perform. 

#### Verification ####
Create two folder with the following contents:
```
derek@degion:~$ ls ./rke2-artifacts
rke2-images.linux-amd64.tar.zst  rke2.linux-amd64.tar.gz  sha256sum-amd64.txt

derek@degion:~$ ls ./rke2-artifacts-with-extra/
rke2-images-canal.linux-amd64.tar.zst   rke2-images-harvester.linux-amd64.tar.gz  rke2.linux-amd64.tar.gz
rke2-images-flannel.linux-amd64.tar.gz  rke2-images.linux-amd64.tar.zst           sha256sum-amd64.txt
```

Run the installer on both:
```
root@degion:/home/derek/rancher/rke2# INSTALL_RKE2_ARTIFACT_PATH=/home/derek/rke2-artifacts sh install.sh
[INFO]  staging local checksums from /home/derek/rke2-artifacts2/sha256sum-amd64.txt
[INFO]  staging zst airgap image tarball from /home/derek/rke2-artifacts/rke2-images.linux-amd64.tar.zst
[INFO]  staging tarball from /home/derek/rke2-artifacts/rke2.linux-amd64.tar.gz
[INFO]  verifying airgap tarball
[INFO]  installing airgap tarball to /var/lib/rancher/rke2/agent/images
[INFO]  verifying tarball
[INFO]  unpacking tarball file to /usr/local
```

```
root@degion:/home/derek/rancher/rke2# INSTALL_RKE2_ARTIFACT_PATH=/home/derek/rke2-artifacts-with-extra sh install.sh
[INFO]  staging local checksums from /home/derek/rke2-artifacts-with-extra/sha256sum-amd64.txt
[INFO]  staging zst airgap image tarball from /home/derek/rke2-artifacts-with-extra/rke2-images.linux-amd64.tar.zst
[INFO]  staging tarball from /home/derek/rke2-artifacts-with-extra/rke2.linux-amd64.tar.gz
[INFO]  verifying airgap tarball
[INFO]  installing airgap tarball to /var/lib/rancher/rke2/agent/images
[INFO]  Installing airgap image from /home/derek/rke2-artifacts-with-extra/rke2-images-canal.linux-amd64.tar.zst
[INFO]  Installing airgap image from /home/derek/rke2-artifacts-with-extra/rke2-images-flannel.linux-amd64.tar.gz
[INFO]  Installing airgap image from /home/derek/rke2-artifacts-with-extra/rke2-images-harvester.linux-amd64.tar.gz
[INFO]  verifying tarball
[INFO]  unpacking tarball file to /usr/local

```

#### Linked Issues ####
https://github.com/rancher/rke2/issues/6308

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -359,6 +359,17 @@ stage_local_airgap_tarball() {
         cp -f "${INSTALL_RKE2_ARTIFACT_PATH}/rke2-images.${SUFFIX}.tar.gz" "${TMP_AIRGAP_TARBALL}"
         AIRGAP_TARBALL_FORMAT=gz
     fi
+    # Search for each CNI we support and copy it to the airgap directory
+    CNIS="calico canal cilium flannel"
```
Comment (brandond) [inline install.sh:363]: I don't love having the CNI list hardcoded in the install script. Any reason we shouldn't just move or copy `rke2-images.*`?

**Review by brandond — CHANGES_REQUESTED** [review]

File: install.sh
Comment (alknopfler) [inline install.sh:1]: Also, I think line451 should be changed to align it to * placeholder

**Review by alknopfler — CHANGES_REQUESTED** [review]

File: install.sh (hunk 1 above)
Comment (alknopfler) [reply install.sh:363]: also, the path ${INSTALL_RKE2_AGENT_IMAGES_DIR} is available here? IIRC it's created on the install phase (after this line446)

Comment (alknopfler) [reply install.sh:363]: and if you're doing a copy as Brandond suggest the lines 636 and 637 (verify_airgap_tarball and install_airgap_tarball are neeeded ?

**Comment by dereknola** [comment]
I have swapped this out for a single `find` command with nice printing 
```
root@degion:/home/derek/rancher/rke2# INSTALL_RKE2_ARTIFACT_PATH=/home/derek/rke2-artifacts sh install.sh
[INFO]  staging local checksums from /home/derek/rke2-artifacts/sha256sum-amd64.txt
[INFO]  staging zst airgap image tarball from /home/derek/rke2-artifacts/rke2-images.linux-amd64.tar.zst
[INFO]  staging tarball from /home/derek/rke2-artifacts/rke2.linux-amd64.tar.gz
[INFO]  verifying airgap tarball
[INFO]  installing airgap tarball to /var/lib/rancher/rke2/agent/images
[INFO]  installing airgap image from /home/derek/rke2-artifacts/rke2-images-flannel.linux-amd64.tar.gz
[INFO]  installing airgap image from /home/derek/rke2-artifacts/rke2-images-harvester.linux-amd64.tar.gz
[INFO]  installing airgap image from /home/derek/rke2-artifacts/rke2-images-canal.linux-amd64.tar.zst
[INFO]  verifying tarball
[INFO]  unpacking tarball file to /usr/local
```

I moved away from doing it in `stage_local_airgap_tarball` because that temporarily stages the image tarball for checksum validation. The install will not continue if that checksum does not match. We want to maintain this functionality. Instead we now move the other airgap images only if the above image checksum match passes. Since we don't have checksums for theses images, we just assume they are fine.

File: install.sh [hunk 2]
```diff
@@ -453,6 +442,9 @@ install_airgap_tarball() {
         info "decompressing airgap tarball to ${INSTALL_RKE2_AGENT_IMAGES_DIR}"
         gzip -dc "${TMP_AIRGAP_TARBALL}" > "${INSTALL_RKE2_AGENT_IMAGES_DIR}/rke2-images.${SUFFIX}.tar"
     fi
+    # Search for and install additonal rke2 images
+    find "${INSTALL_RKE2_ARTIFACT_PATH}" -type f -name "rke2-images-*.${SUFFIX}*" \
```
Comment (brandond) [inline install.sh:446]: The install script doesn't currently require findutils be installed on the host. Can we do this with just a `for IMAGE in ${INSTALL_RKE2_ARTIFACT_PATH}/rke2-images-*.${SUFFIX}*; do cp -v "${IMAGE}" "${INSTALL_RKE2_AGENT_IMAGES_DIR}" ; done` loop instead?

**Review by brandond — CHANGES_REQUESTED** [review]

File: install.sh (hunk 2 above)
Comment (dereknola) [reply install.sh:446]: Sure

**Review by brandond — APPROVED** [review]

**Review by alknopfler — APPROVED** [review]
LGTM

**Review by vitorsavian — APPROVED** [review]

**Pull request merged**